src/System.Numerics.Vectors/src/System/Numerics/Quaternion.cs

```diff
@@ -100,18 +101,9 @@ public float LengthSquared()
         /// <returns>The normalized Quaternion.</returns>
         public static Quaternion Normalize(Quaternion value)
         {
-            Quaternion ans;
-
-            float ls = value.X * value.X + value.Y * value.Y + value.Z * value.Z + value.W * value.W;
-
-            float invNorm = 1.0f / MathF.Sqrt(ls);
-
-            ans.X = value.X * invNorm;
-            ans.Y = value.Y * invNorm;
-            ans.Z = value.Z * invNorm;
-            ans.W = value.W * invNorm;
-
-            return ans;
+            Vector4 q = Unsafe.As<Quaternion, Vector4>(ref value);
+            q = Vector4.Normalize(q);
+            return Unsafe.As<Vector4, Quaternion>(ref q);
```

Comment:
Not sure if `Unsafe.As<Vector4, Quaternion>(ref q)` will prevent `q` being a register (e.g. address taken)

e.g. should it be
```csharp
public static Quaternion Normalize(Quaternion value)
{
    Vector4 q = Unsafe.As<Quaternion, Vector4>(ref value);
    Vector4 result = Vector4.Normalize(q);
    return Unsafe.As<Vector4, Quaternion>(ref result);
}
```